Factory/DiactorosFactory.php

```diff
@@ -55,7 +55,7 @@ public function createRequest(Request $symfonyRequest)
         $request = new ServerRequest(
             $server,
             DiactorosRequestFactory::normalizeFiles($this->getFiles($symfonyRequest->files->all())),
-            $symfonyRequest->getUri(),
+            $symfonyRequest->getSchemeAndHttpHost().$symfonyRequest->getRequestUri(),
```

Comment:
I've done this defensively to avoid repeats of the problem I have encountered.

[`getUri`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1118) creates a URL in a very indirect way. This will likely be a problem due to the way Symfony requests normalise different parts of the URL. In comparison, [`getRequestUri`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1089) is much more direct, you can see in [`prepareRequestUri`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1775) how the URL comes directly from the superglobals or headers.

As we're dealing with a conversion between request types, we need to use the cleanest data we can when constructing the PSR7 request. Otherwise we can get a URL different to the original